perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java

```diff
@@ -953,6 +989,19 @@ public interface ServicesManager {
 	 */
 	void removeDestination(PerunSession perunSession, Service service, Facility facility, Destination destination) throws PrivilegeException, ServiceNotExistsException, FacilityNotExistsException, DestinationAlreadyRemovedException;
 
+	/**
+	 * Removes list of rich destinations, each destination is removed from the rich destination's facility and service.
+	 *
+	 * @param perunSession
+	 * @param richDestinations list of rich destinations
+	 * @throws PrivilegeException
+	 * @throws InternalErrorException
+	 * @throws ServiceNotExistsException
+	 * @throws FacilityNotExistsException
+	 * @throws DestinationAlreadyRemovedException
+	 */
+	void removeRichDestinations(PerunSession perunSession, List<RichDestination> richDestinations) throws PrivilegeException, ServiceNotExistsException, FacilityNotExistsException, DestinationAlreadyRemovedException;
```

Comment:
Just a nitpick: technically we are not removing RichDestinations but Destinations, something like 'removeDestinationsByRichDestinations' would be more correct ('removeDestinations' would be fine by me too anyways}